src/main/scala/com/scality/clueso/SparkUtils.scala

```diff
@@ -0,0 +1,129 @@
+package com.scality.clueso
+
+import java.io.{File, IOException}
+import java.net.URI
+
+import com.scality.clueso.query.{MetadataQuery, MetadataQueryExecutor}
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.hadoop.conf.{Configuration => HadoopConfig}
+import org.apache.hadoop.fs.{FileStatus, FileSystem, Path, PathFilter}
+import org.apache.spark.sql.functions.{col, lit}
+import org.apache.spark.sql.{Row, SparkSession}
+
+
+object SparkUtils extends LazyLogging {
+  def loadCluesoConfig(confFilePath: String) = {
+    val parsedConfig = ConfigFactory.parseFile(new File(confFilePath))
+    val _config = ConfigFactory.load(parsedConfig)
+
+    new CluesoConfig(_config)
+  }
+
+  def buildHadoopFs(config: CluesoConfig) = {
+    FileSystem.get(new URI(PathUtils.landingURI(config)), hadoopConfig(config))
+  }
+
+  def hadoopConfig(config: CluesoConfig): HadoopConfig = {
+    val c = new HadoopConfig()
+    c.set("fs.s3a.connection.ssl.enabled", config.s3SslEnabled)
+    c.set("fs.s3a.endpoint", config.s3Endpoint)
+    c.set("fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+    c.set("fs.s3a.access.key", config.s3AccessKey)
+    c.set("fs.s3a.secret.key", config.s3SecretKey)
+    c.set("fs.s3a.path.style.access", config.s3PathStyleAccess)
+    c.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
```

Comment:
have you noticed any change since setting this?

Reply:
does this need to be here or just in the sparkSession?

Reply:
Now it's defined on spark-defaults.conf in livy:/spark/conf . I cannot yet say how good it is, will do a simulation with the 7m records like before to do an apples to apples comparison